node/src/proxy_server/mod.rs

```diff
@@ -3793,7 +3849,7 @@ mod tests {
                 assert!(!proxy_server.stream_key_routes.is_empty());
                 assert!(!proxy_server.tunneled_hosts.is_empty());
                 TestLogHandler::new().exists_log_containing(&format!(
-                    "DEBUG: {test_name}: Last data received for stream key vYMR621EYFzvxTJqpzrGKr3KMbc, \
+                    "DEBUG: {test_name}: Last data received for stream key zpi6SfvohvBgqq9y8zLgzIhxPS4, \
```

Comment:
Instead of having the literal value of `stream_key` here, can you make this assert refer to the variable `stream_key` itself, so that if the method of making a meaningful stream key changes in the future this test won't fail?

Also line 3870 below.

Reply:
It seems that extracting the other values may not be feasible at this point.

```rust
match self
    .stream_key_routes
    .get(&stream_key)
    .unwrap()
    .expected_services
{
    ExpectedServices::OneWay(expected_services) => match expected_services[0] {
        ExpectedService::Routing(_, _, _) => {}
        ExpectedService::Exit(_, _, _) => {}
        ExpectedService::Nothing => {}
    },
    ExpectedServices::RoundTrip(_, _, _) => {}
}
```